neurox/analysis/plotting.py

```diff
@@ -84,3 +84,172 @@ def plot_accuracies(title, overall_acc,
+        fig, ax = plt.subplots(4)
+        ax[0].plot(model_weights_1)
+        ax[0].set(title = 'Mean distribution of weights')
+        ax[1].plot(model_weights_3.transpose())
+        ax[1].set(title ='Mean weight per layer')
+        ax[2].plot(sorted(model_weights_1))
+        ax[2].set(title = 'Sorted weights')
+        plt.show()
+        sort_idx = np.argsort(model_weights)[::-1]
+        print("\nNeurons with most weight : "+str(sort_idx[:10]))
+    else :
+        if (model_name=='SGL') : 
+            model_weights = kwargs['weights']
+        elif (model_name=='NeuroX') :
+            model_weights = get_weights(kwargs['model'], model_name)
+        
+        
+        plt.rcParams["figure.figsize"] = (10,12)
+        fig, ax = plt.subplots(3)
+        ax[0].plot(model_weights[label2idx[label]])
+        ax[0].set(title = 'Signed weights for label '+label, xlabel = 'Neurons', ylabel = 'Neuron weights')
+        ax[1].plot(np.abs(model_weights[label2idx[label]]))
+        ax[1].set(title ='Absolute weights for label '+label, xlabel = 'Neurons', ylabel = 'Neuron weights')
+        ax[2].plot(sorted(np.abs(model_weights[label2idx[label]])))
+        ax[2].set(title ='Sorted weights for label '+label)
+        fig.tight_layout(pad = 2.0)
+        plt.show()  
+        
+def plot_weight_distribution_per_layer(model_name, data, label, layer, **kwargs) : 
+    label2idx = data["label2idx"]
+    if (model_name=='SGL') :
+        model_weights = kwargs['weights']
+    elif (model_name=='NeuroX') :
+        model_weights = get_weights(kwargs['model'], model_name)
+   
+        
+    if (label=='total') : 
+        weights = np.mean(np.abs(model_weights), axis = 0)
+    else :
+        weights = np.abs(model_weights[label2idx[label]])
+        print("For label "+label)
+        
+    weights = np.reshape(weights, (9984,1))
+    weights_layerwise = np.split(weights, 13, axis=0)
+        
+    if (layer==0) : 
+        plt.rcParams["figure.figsize"] = (15,15)
+        fig, ax = plt.subplots(4,4)
+        for i in range(0,13) :
+            ax[int(i/4), int(i%4)].plot(weights_layerwise[i])
+            ax[int(i/4), int(i%4)].set(title = 'Layer '+str(i+1)+' weights')
+            mean = weights_layerwise[i].mean()
+            sparsity_mask = weights_layerwise[i] > 1e-10 * mean
+            print("\nFeatures selected from Layer "+str(i+1)+" : "+str(sparsity_mask.sum()))
+            print("Spiky Neuron : "+str(np.argmax(weights_layerwise[i]))) 
+        fig.tight_layout(pad= 2.0)
+        plt.show()
+            
+    else : 
+        plt.rcParams["figure.figsize"] = (5,5)
+        mean = weights_layerwise[layer-1].mean()
+        sparsity_mask = weights_layerwise[layer-1] > 1e-10 * mean
+        print("Features selected from Layer "+str(layer)+" : "+str(sparsity_mask.sum()))
+        plt.plot(weights_layerwise[layer-1])
+        plt.title('Layer '+str(layer)+' weights')
+        plt.show()
+        
+def sparsity(model_name, data, **kwargs) :
```

Comment:
This and the following functions seem "internal" to the lib, can you prepend them with an underscore so the name becomes `_sparsity`? Also it might be better to name it to something a bit more clear, is this plotting sparsity? Computing it?